packages/@aws-cdk/aws-sns-subscriptions/lib/lambda.ts

```diff
@@ -36,6 +36,12 @@ export class LambdaSubscription implements sns.ITopicSubscription {
       principal: new iam.ServicePrincipal('sns.amazonaws.com'),
     });
 
+    // if the topic and function are created in different stacks
+    // then we need to make sure the topic is created first
+    if (topic.stack !== this.fn.stack) {
```

Comment:
Is it possible this topic is an imported (`ITopic`)? If so, is this necessary?

My spidey sense tingles hard whenever we add a dependency like this; I feel like these types of aides lead to circular dependency bugs that aren't obvious and don't come to light until the next release. Just want to limit the blast radius to as small a use case as possible.

Reply:
That's a good point. I could add a type check here as well. I did run into issues when testing this if the dependency is not explicit, the deployment would fail because the function (or sqs) stack was getting created before the topic stack.

Reply:
updated.